articles/150_roslaunch.md

```diff
@@ -0,0 +1,364 @@
+In the future this might change, so reference the design doc[^lifecylce] or future documentation on the subject.
+
+#### Runtime
+
+During runtime, a Managed ROS node emits events anytime the state of the node changes.
+This is at least emitted on a topic, but could also be captured, aggregated, and/or communicated in other ways too.
+These state changes could be consumed by either the launch system itself or by the user, either of which could react to these changes.
+
+For example, the user could express something like "when node 'A' enters the `Active` state, launch nodes 'B' and 'C'" or "if node 'A' exits with a return code or enters the `Finalized` state, shutdown everything".
+
+<div class="alert alert-warning" markdown="1">
+RFC:
+
+This is where things get murky for me.
+There is a gray line between more complex startup behaviors (like "wait for A to get to Active then launch B") and long running monitoring and reactions to events, what I would call "supervision" (like "if A goes transitions to ErrorProcessing change B to Inactive").
+
+I'm interested to see what other people think.
+
+For me the best way to separate the two cases (if they're to be separated at all) is to ask "do I want/need to understand this event handling off-line"?
+For example, when looking at a system description, it would be nice to see the event handler "when A crashes or exits, the whole system is down" (i.e. `require=true`) or the event handler "when A reaches Inactive, launch B" (something like "B depends on A").
+However, maybe it doesn't need to capture/understand the "when A crashes or has an error move B to inactive".
+
+Alternatively, we could say "all event handling is opaque to the launch system and is user defined" which removes the ambiguity.
+
+\<wjwwod's opinion>
+<br/>
+I think the idea that saying all "supervision" or "reactionary launching" like activities are not part of the launch system is the simplest way forward, but then you'd still like to include in the launch system some convenient ways to express the most common cases, e.g. something like ROS 1's `roslaunch`'s `require=true` and `respawn=true`.
+In my opinion, it would be unacceptably inconvenient to have users write a lambda for each of those cases, and I also think it would lead to everyone doing it slightly different and making it impossible to understand with tools.
+<br/>
+</wjwwod's opinion>
+
+This will come up again in the "System Description" section, because I believe the mind set around the "Execution of the System Description" (basically the python library that runs things) will be "personal responsibility".
+</div>
+
+#### Termination
+
+Managed ROS Nodes have some additional observable effects when terminating (the node, not necessarily the process containing it).
+A managed node enters the `Finalized` state after passing through the `ShuttingDown` transition state on termination.
```

Comment:
During termination will ROS 2 launch try to transition to the Finalized state, or just send signals and hope handling code in the node takes care of it?

Reply:
A good question, to which I do not know the answer. My gut reaction would be to say it sends a signal and the launch system tries not to send state transition requests itself, unless directed to do so by the user. So maybe, defaults to signals, the user can ask the launch system to transition first, then signal. I'll mention it and add an RFC block about the topic.

Reply:
It's possible, and I do expect the launch system will be a Python ROS node itself, though it might not be a managed one to start with, partially because there's no lifecycle in Python atm and because it's simpler to start without that.

But I can imagine calling the launch system from the launch system in a nesting fashion. This just has to be balanced with including other launch files instead. I think that might be more transparent than calling the launch systems executable multiple times.

Reply:
For instance, in ROS 1, the capability server is often called by roslaunch and itself in turn calls roslaunch multiple times. If I were to reimplement that in ROS 2, I would hope to instead have the capability server ask an existing launch system to add something dynamically, rather than starting a new instance. That is if the capability server is not an instance of the launch system itself or rolled into it entirely.

Reply:
I added a note that this is not specified in this document, basically kicking the can down the road to some other lifecycle related documentation: 7d6c5fa